libp2p/services/hpservice.nim

```diff
@@ -71,22 +96,10 @@ method setup*(self: HPService, switch: Switch): Future[bool] {.async.} =
     let dcutrProto = Dcutr.new(switch)
     switch.mount(dcutrProto)
 
-    self.newConnectedPeerHandler = proc (peerId: PeerId, event: PeerEvent): Future[void] {.async.} =
-      try:
-        let conn = switch.connManager.selectMuxer(peerId).connection
-        if isRelayed(conn) and conn.transportDir == Direction.In:
-          if await self.tryStartingDirectConn(switch, peerId):
-            await conn.close()
-            return
-          let dcutrClient = DcutrClient.new()
-          var natAddrs = switch.peerStore.getMostObservedProtosAndPorts()
-          if natAddrs.len == 0:
-            natAddrs =  switch.peerInfo.listenAddrs.mapIt(switch.peerStore.guessDialableAddr(it))
-          await dcutrClient.startSync(switch, peerId, natAddrs)
-          await sleepAsync(2000.milliseconds) # grace period before closing relayed connection
-          await conn.close()
-      except CatchableError as err:
-        debug "Hole punching failed during dcutr", err = err.msg
+    proc newConnectedPeerHandler(peerId: PeerId, event: PeerEvent) {.async.} =
+      await newConnectedPeerHandlerAux(self, switch, peerId, event)
+
+    self.newConnectedPeerHandler = newConnectedPeerHandler
```

Comment:
```suggestion
    self.newConnectedPeerHandler = proc (peerId: PeerId, event: PeerEvent) {.async.} =
      await newConnectedPeerHandlerAux(self, switch, peerId, event)
```
And you cane rename newConnectedPeerHandlerAux to onNewPeer or something like that